trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/RequestBuilder.java

```diff
@@ -36,12 +38,28 @@
 
 class RequestBuilder
 {
+    private static final Logger log = Logger.get(RequestBuilder.class);
+
     private RequestBuilder() {}
 
     static Request fromRequest(ContainerRequest request)
     {
+        MultivaluedMap<String, String> requestHeaders = request.getHeaders();
+        String xAmzDate = Optional.ofNullable(requestHeaders.getFirst("x-amz-date")).orElseThrow(() -> {
+            log.debug("Missing \"x-amz-date\" header");
+            return new WebApplicationException(Response.Status.BAD_REQUEST);
+        });
+        Optional<String> session = Optional.ofNullable(requestHeaders.getFirst("x-amz-security-token"));
```

Comment:
Oh I see - this is the security token header. Let's use a different name than `session` please. I got confused with the session in the credential. I think a name that signals this is from the request header would be more comprehensible.

Reply:
Good point, will rename it